content_api/new_upload/readers/script.py

```diff
@@ -0,0 +1,197 @@
+import hashlib
+import logging
+import re
+
+from ..exceptions import ValidationException
+from ...helpers.enums import PackageType
+
+log = logging.getLogger(__name__)
+
+
+class EntryScript:
+    """
+    Entry script meta data.
+
+    @ivar md5sum: md5 checksum of script file
+    @type md5sum: C{bytes}
+
+    @ivar unique_id: Unique ID of script (aka short_name)
+    @type unique_id: C{str}
+    """
+
+    def __init__(self):
+        self.md5sum = None
+        self.unique_id = None
+        self.package_type = None
+
+    def read(self, fp):
+        """
+        Read an entry script ("info.nut" or "library.nut").
+
+        @param fp: Filepointer to read (should already be open)
+        @type fp: File-like object
+
+        @return: True on success.
+        @rtype: C{bool}
+        """
+
+        md5sum = hashlib.md5()
+
+        # Bit of Python trickery to allow reading the file only once,
+        # while running two completely independent functions with
+        # internal state. This makes the functions easier to understand,
+        # while not caching the content of the file in a variable.
+        unique_id = self._read_unique_id()
+        script_type = self._read_script_type()
+        next(unique_id)
+        next(script_type)
+
+        # Check the BOM if we should decode in UTF-8 or latin-1
+        line = fp.readline()
+        md5sum.update(line)
+        decode_utf8 = False
+        if len(line) >= 3:
+            # Check for BOM marker for UTF-8; both in LE and BE.
+            # Python will take care of the endianess.
+            if (line[0] == 0xEF and line[1] == 0xBB) or (line[0] == 0xBB and line[1] == 0xEF):
```

Comment:
this doesn't seem quite right - UTF8 without a BOM is much more common than UTF8 with a BOM, yet here you're only decoding it as UTF8 if it has a BOM. Would not just trying a `line.decode()` with a fallback of `line.decode("latin-1")` if it errors be enough?